chrome/word-manager.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Handles the parsing of the page text as well as saving the words.
+ *
+ * @param {string} text
+ * @param {string} language
+ * @constructor
+ */
+WordManager = function(text, language) {
+  this.text_ = text;
+  this.language_ = language;
+  this.localDb_ = new LocalDb();
+};
+
+/**
+ * Parses the text and saves the resulting words.
+ */
+WordManager.prototype.processWords = function() {
+  if (this.language_ != 'en' && this.language_ != 'uk') {
+    throw new Error('Language not supported');
+  }
+
+  wordsFromText = this.parseWords_();
+  this.saveWords_(wordsFromText);
+};
+
+/**
+ * Parses the given text by blank spaces and separators, returning a lower case word array.
+ * @returns {Array}
+ */
+WordManager.prototype.parseWords_ = function() {
+  var splitText = this.text_.split(/[\u2000-\u206F\u2E00-\u2E7F\\'!"#\$%&\(\)\*\+,\-\.\/:;<=>\?@\[\]\^_`\{\|\}~\s]/);
+	
+  var words = [];
+	
+  for (var i = 0; i < splitText.length; i++) {
+    if (splitText[i] != "") {
+      words.push(splitText[i].toLowerCase());
+    }
+  }
+  
+  return words;
+};
+
+/**
+ * Returns the language's name in the Lang enum.
+ * @returns {Lang}
+ */
+WordManager.prototype.getLanguageName_ = function() {
+  if(this.language_ == 'en') {
```

Comment:
consider having an object mapping = {
  'en': Lang.ENGLISH,
  'uk': Lang.Ukrainian,
 ...
}

and than mapping[language] and ensure it's not null.